thirdparty/DiskANN/src/math_utils.cpp

```diff
@@ -221,9 +220,12 @@ namespace math_utils {
 
     bool is_norm_given_for_pts = (pts_norms_squared != NULL);
 
-    float* pivs_norms_squared = new float[num_centers];
-    if (!is_norm_given_for_pts)
-      pts_norms_squared = new float[num_points];
+    auto pivs_norms_squared = std::make_unique<float>(num_centers);
+    std::unique_ptr<float[]> pts_norms_squared_deleter = nullptr;
```

Comment:
Use deleter approach as `pts_norms_squared` has default value in the parameter list